Khan/perseus
# [Locked Figure Aria] Fix $ edge cases for spoken math aria labels

## Summary:
I received a couple suggestions in PR https://github.com/Khan/perseus/pull/1854.
I'm implmementing them here:

- Wrote a helper function that joins labels together so that we don't have to
  duplicate that code for every single locked figure.
- Stopped calling `generateSpokenMathDetails()` on the whole aria string
  for the locked figure. Instead, it's now being called within
  the new helper function so that the labels can individually be converted,
  and we don't have to worry about the edge case with two different labels
  having dollar signs and causing all the content in between to be counted
  as TeX when it shouldn't.
- Handled the case with a lone unescaped dollar sign, which should not count
  as TeX.

Note: The lone unescaped dollar isn't handled for the TeX inside the
graph itself yet. This will be in the next PR.

Issue: https://khanacademy.atlassian.net/browse/LEMS-2548

## Test plan:
`yarn jest`

Storybook
- http://localhost:6006/?path=/story/perseuseditor-widgets-interactive-graph--mafs-with-locked-figure-labels-all-flags
- Add two visible labels to the point, with one having "$1 and" as the label
  and the other having "$1" as the label.
- Press the autogenerate button and confirm that the "and" is normal and not "a n d".
  Also confirm that the dollar signs show up normally.
- Add two TeX labels and confirm that the auto-generated text is as expected.
- Repeat for all the other locked figures.
